BUG: Raise TypeError for scalar indexer in take

gfkang opened the pull request:
- [x] closes #42875
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

gfkang committed:
BUG: Raise TypeError for scalar indexer in take

phofl commented on pandas/core/indexes/base.py:
```diff
@@ -1059,6 +1059,8 @@ def take(
         if kwargs:
             nv.validate_take((), kwargs)
+        if np.isscalar(indices):
+            raise TypeError("Expected indices to be array-like")
```
Please use is_scalar from pandas.

phofl commented on pandas/tests/indexes/test_indexing.py:
```diff
@@ -70,6 +70,13 @@ def test_take(self, index):
                 index.freq
 
+    def test_take_indexer_type(self):
+        integer_index = Int64Index([0, 1, 2, 3])
```
I think this index constructor is deprecated, please use Index(

phofl commented on pandas/tests/indexes/test_indexing.py:
```diff
@@ -70,6 +70,13 @@ def test_take(self, index):
             with pytest.raises(AttributeError, match=msg):
                 index.freq
 
+    def test_take_indexer_type(self):
```
Please add issue number

phofl reviewed (changes requested):
Couple of comments. Also please add whatsnew

gfkang committed:
Make requested changes
Use pandas is_scalar, remove deprecated constructor, add GH issue number to test

gfkang committed:
Add whatsnew entry

phofl commented:
Could you rebase master? We had a few ci failures

mroeschke commented:
Thanks @gfkang
